ntcore/doc/time-sync-protocol.adoc

```diff
@@ -0,0 +1,53 @@
+= Time Syncronization Protocol Specification, Version 1.0
+WPILib Developers <wpilib@wpi.edu>
+Protocol Revision 1.0, 08/25/2024
+:toc:
+:toc-placement: preamble
+:sectanchors:
+
+[[roles]]
+=== Roles
+
+Roles assumed by Time Syncronization Protocol (TSP) participants consists of exactly one server and zero or more clients per network. The server role is responsible for listening for incoming time synchronization requests from clients and replying approriately. The client role is responsible for sending "Ping" messages to the server and listening for "Pong" replies to estimate the offset between the server and client time bases. 
```

Comment:
```suggestion
Time Syncronization Protocol (TSP) participants can assume either a server role or a client role. Each network consists of exactly one server and zero or more clients. The server role is responsible for listening for incoming time synchronization requests from clients and replying approriately. The client role is responsible for sending "Ping" messages to the server and listening for "Pong" replies to estimate the offset between the server and client time bases.
```
Might just be personal preference, but it seems clearer to separate the statements about the roles of partipants and about the number of each role in a network.

Reply:
We should probably also define what "network" means